### PR TITLE
Update third-party dependencies - JS #6193

### LINKED ISSUE #6193: Update third-party dependencies - JS
Same as #6192 but for JS.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #6193 

All dependencies are updated to the latest stable except for the following:
- Bootstrap: it's an extensively-used front-end framework and its changes should be manually eyeballed.
- jQuery: updated to the latest `1.x.x` version; changing the major version is an issue on its own #3806.
- topojson, d3: cannot change major version due to dependency from datamaps.

### CODE REVIEW

**Comment by damithc** [comment]
Anything current devs need to do (e.g. install new versions of tools) after pulling this patch?

**Comment by wkurniawan07** [comment]
@damithc JS dependencies are all checked in, so nothing needs to be done.

**Comment by wkurniawan07** [comment]
@JosephineKwa let's review this?

File: static-analysis/teammates-eslint.yml [hunk 1]
```diff
@@ -40,12 +40,12 @@ rules:
     # disallow duplicate case labels
     no-duplicate-case: "error"
 
-    # disallow empty block statements
-    no-empty: "error"
-
     # disallow empty character classes in regular expressions
     no-empty-character-class: "error"
 
+    # disallow empty block statements
```
Comment (JosephineKwa) [inline static-analysis/teammates-eslint.yml]: Hm Just wondering why was this changed though?

Comment (wkurniawan07) [inline static-analysis/teammates-eslint.yml]: I upgraded ESLint to 3.11.1 and the ruleset needs to be updated. In some places I re-ordered the rules as found on [this page](http://eslint.org/docs/rules/). Sorry for the confusion.

File: src/main/webapp/js/common.js [hunk 2]
```diff
@@ -772,7 +775,7 @@ function isInstitutionValid(rawInstitution) {
         return false;
     }
     
-    if (institution.match(/[^\/\\,.'\-\(\)0-9a-zA-Z \t]/)) {
+    if (institution.match(/[^/\\,.'\-()0-9a-zA-Z \t]/)) {
```
Comment (damithc) [inline src/main/webapp/js/common.js]: Is this regex duplicated in line 753? In that case we can define it once in some appropriate place (may be as a `FirstTimer` issue)

File: src/main/webapp/js/index.js [hunk 3]
```diff
@@ -7,15 +7,16 @@ var OPERATION_STUDENT_LOGIN = 'student_login';
...
+    'I just wanted to let you know that TEAMMATES has been a great success!  Students love it. <br>-Faculty user, USA',
+    'I had such a great experience with TEAMMATES in the previous semester that I am back for more! <br>-Faculty user, Pakistan',
+    'Thank you for this. I think it is brilliant. <br>-Faculty user, Canada',
+    'I found the TEAMMATES system really easy to use. On the whole a very positive experience. Using TEAMMATES certainly helps with one of the main potential problems of group-based assessments. <br>-Faculty user, Singapore',
+    'I find it really great and so simple to use. <br>-Faculty user, Austria',
+    'These peer evaluations will be perfect for classes.  I can already see that this is going to be an excellent tool as I need the teams to evaluate each other on a weekly basis.  Adding a new evaluation item and the questions/response criteria is so easy through your system. <br>-Faculty user, USA',
+    'Thank you for building such a wonderful tool. <br>-Faculty user, Canada'
```
Comment (damithc) [inline src/main/webapp/js/index.js]: Possible to use line wrapping? If yes, can be packaged into a FirstTimer issue.
